core/advancedpolls.php

```diff
@@ -346,6 +352,23 @@ public function do_poll_modification($topic_data)
 	}
 
 	/**
+	 * Add the possible options to the template
+	 *
+	 * @param array	$post_data		The array of post data
+	 * @return void
+	 */
+	public function do_poll_block_modification($poll_options_template_data)
+	{
+		usort($poll_options_template_data, array($this, "order_by_votes"));
+		return $poll_options_template_data;
+	}
+
+	protected function order_by_votes($a, $b)
+	{
+		return (((int) $a['POLL_OPTION_RESULT'] < (int) $b['POLL_OPTION_RESULT']) ? 1 : (((int) $a['POLL_OPTION_RESULT'] > (int) $b['POLL_OPTION_RESULT']) ? -1 : (((int) $a['POLL_OPTION_ID'] > (int) $b['POLL_OPTION_ID']) ? 1 : -1))); 
```

Comment:
Not so much simpler your way, should be:

return (((int) $b['POLL_OPTION_RESULT'] != (int) $a['POLL_OPTION_RESULT'])
? ((int) $b['POLL_OPTION_RESULT'] - (int) $a['POLL_OPTION_RESULT']) :
((int) $a['POLL_OPTION_ID'] - (int) $b['POLL_OPTION_ID']) );

In case of same number of votes, the order should be by option id.


2015-04-07 4:08 GMT+02:00 Wolfsblvt <notifications@github.com>:

> In core/advancedpolls.php
> <https://github.com/Wolfsblvt/advancedpolls/pull/17#discussion_r27847158>:
>
> > @@ -346,6 +352,23 @@ public function do_poll_modification($topic_data)
> >  	}
> >
> >  	/**
> > +	 * Add the possible options to the template
> > +	 *
> > +	 * @param array	$post_data		The array of post data
> > +	 * @return void
> > +	 */
> > +	public function do_poll_block_modification($poll_options_template_data)
> > +	{
> > +		usort($poll_options_template_data, array($this, "order_by_votes"));
> > +		return $poll_options_template_data;
> > +	}
> > +
> > +	protected function order_by_votes($a, $b)
> > +	{
> > +		return (((int) $a['POLL_OPTION_RESULT'] < (int) $b['POLL_OPTION_RESULT']) ? 1 : (((int) $a['POLL_OPTION_RESULT'] > (int) $b['POLL_OPTION_RESULT']) ? -1 : (((int) $a['POLL_OPTION_ID'] > (int) $b['POLL_OPTION_ID']) ? 1 : -1)));
>
> Easier statement would be:
>
> return ((int) $b['POLL_OPTION_RESULT'] - (int) $a['POLL_OPTION_RESULT']))
>
> For usort the it doesn't matter if you return 1 or 5, that just means $b
> is higher.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Wolfsblvt/advancedpolls/pull/17/files#r27847158>.
>